### PR TITLE
Adds TrustedHTML, TrustedScript, TrustedScriptURL

### PR DESCRIPTION
**Pull request by rachelandrew**
Working on documenting the Trusted Types API. This PR adds:

- TrustedHTML
- TrustedScript
- TrustedScriptURL

Spec: https://w3c.github.io/webappsec-trusted-types/dist/spec/

Reviewer: @jpmedley 

To save yourself some time Joe, these are all pretty much the same as each other so if you want to snag TrustedHTML and the two method pages, I will then apply any changes to the others.

Also, it looks as if toJSON() is only in Canary (I tested it), and we have no BCD for the method. So that will need adding, which I can do if it is going to land.

### CODE REVIEW

File: files/en-us/web/api/trustedhtml/index.html [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+slug: Web/API/TrustedHTML
+tags:
+  - API
+  - Interface
+  - Reference
+  - TrustedHTML
+---
+<div>{{DefaultAPISidebar("Trusted Types API")}}</div>
+
+<p class="summary">The <strong><code>TrustedHTML</code></strong> interface of the {{domxref('Trusted Types API')}} represents a string that a developer can insert into an <a href="/en-US/docs/Web/API/Trusted_Types_API#injection_sinks">injection sink</a> that will render it as HTML. These objects are created via {{domxref("TrustedTypePolicy.createHTML")}} and therefore have no constructor.</p>
```
Comment (jpmedley) [inline files/en-us/web/api/trustedhtml/index.html:12]:
```suggestion
<p class="summary">The <strong><code>TrustedHTML</code></strong> interface of the {{domxref('Trusted Types API')}} represents a string that a developer can insert into an <a href="/en-US/docs/Web/API/Trusted_Types_API#injection_sinks">injection sink</a> that will render it as HTML. These objects are created via {{domxref("TrustedTypePolicy.createHTML", "TrustedTypePolicy.createHTML()")}} and therefore have no constructor.</p>
```

Comment (jpmedley) [reply files/en-us/web/api/trustedhtml/index.html:12]: I can't remember if the macro prints the '()' when it's a method.

File: files/en-us/web/api/trustedhtml/index.html [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+<p>A <strong>TrustedHTML</strong> object has a <code>[[Data]]</code> internal slot which stores a {{domxref("DOMString")}}. The value is set when the object is created and cannot be changed by JavaScript as there is no setter exposed.</p>
```
Comment (jpmedley) [inline files/en-us/web/api/trustedhtml/index.html:14]: I'm not crazy about explaining APIs in terms of what happens under the hood. Inputs and results are what I think these need to be about.

File: files/en-us/web/api/trustedhtml/index.html [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+<h2 id="Methods">Methods</h2>
+
+<dl>
+  <dt>{{domxref("TrustedHTML.toJSON()")}}</dt>
+  <dd>A {{domxref("DOMString")}}  containing the value of the <code>[[Data]]</code> internal slot.</dd>
```
Comment (jpmedley) [inline files/en-us/web/api/trustedhtml/index.html:20]:
```suggestion
  <dd>Returns a JSON representation of the stored data.</dd>
```

File: files/en-us/web/api/trustedhtml/index.html [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  <dt>{{domxref("TrustedHTML.toString()")}}</dt>
+  <dd>A {{domxref("DOMString")}} containing the value of the <code>[[Data]]</code> internal slot.</dd>
```
Comment (jpmedley) [inline files/en-us/web/api/trustedhtml/index.html:22]: Do something similar here.

File: files/en-us/web/api/trustedhtml/index.html [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+</dl>
+
+<h2 id="Examples">Examples</h2>
```
Comment (jpmedley) [inline files/en-us/web/api/trustedhtml/index.html:25]: Please provide a short description of what the example is showing.

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/trustedhtml/index.html (hunk 2 above)
Comment (rachelandrew) [reply files/en-us/web/api/trustedhtml/index.html:14]: Yep that's fine.

File: files/en-us/web/api/trustedhtml/index.html (hunk 3 above)
Comment (jpmedley) [reply files/en-us/web/api/trustedhtml/index.html]: JavaScript has no concept of a DOMString. That's an IDL notion. All it has is a string. Changing that universally is outside the scope of this. What I'm getting at is that the string that's returned is structured as JSON and other places where that occurs we say say as much.

Comment (rachelandrew) [reply files/en-us/web/api/trustedhtml/index.html]: seems odd when it is literally just returning a string, but ok.

**Review by jpmedley — APPROVED** [review]

File: files/en-us/web/api/trustedhtml/tojson/index.html [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  - toJSON
+  - TrustedHTML
+---
+<div>{{DefaultAPISidebar("Trusted Types API")}}</div>
+
+<p class="summary">The <strong><code>toJSON()</code></strong> method of the {{domxref("TrustedHTML")}} interface returns a JSON representation of the stored data.</p>
+
+<h2 id="Syntax">Syntax</h2>
+
+<pre class="syntaxbox">var <var>json</var> = <var>TrustedHTML</var>.toJSON();</pre>
```
Comment (foolip) [inline files/en-us/web/api/trustedhtml/tojson/index.html:17]: Sorry to poke an old PR, but this was linked from https://github.com/mdn/browser-compat-data/pull/11063 and I wanted to bring attention to https://github.com/mdn/content/issues/3487.

https://developer.mozilla.org/en-US/docs/Web/API/TrustedHTML/toJSON now has `var json = TrustedHTML.toJSON()` in the syntax block, but `TrustedHTML.toJSON()` is not a piece of code that works. `TrustedHTML` would need to be some plausible name for an instance of `TrustedHTML`, so perhaps `escaped.toJSON()`?

**Pull request merged**